### ISSUE TITLE
make does recompile everything, even without changes

### ISSUE THREAD

**Issue by NobbZ** [idris-lang/Idris-dev#1821]
I wanted to contribute a small piece of code (#1071, a low hanging fruit, not ready yet).

After every change I do recompile idris using `make`, which seems to rebuild the whole project, even if only one or two source files changed.

This takes quite an huge amount of time (~5 minutes) on my machine:

```
$ uname -s -r -v -m -p -i -o
Linux 3.13.0-43-generic #72-Ubuntu SMP Mon Dec 8 19:35:06 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux

$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.8.3
```

CPU: Intel® Core™ i5-4210M CPU @ 2.60GHz × 4
RAM: 8 GiB

**Comment by edwinb** [comment]
I rebuild using “cabal install” rather than make (although that’s all ‘make’ does), and this doesn’t happen (it would make development essentially impossible if it did, after all). Which files are you editing?

Edwin.

**Comment by NobbZ** [comment]
In the worst case? Not even a single one and cabal decides to recompile the
whole lot of stuff.

Anyway, so far I modified repl, option AST and main (don't remember exact
names because I'm on mobile.
Am 02.01.2015 14:58 schrieb "Edwin Brady" <notifications@github.com>:

> I rebuild using “cabal install” rather than make (although that’s all
> ‘make’ does), and this doesn’t happen (it would make development
> essentially impossible if it did, after all). Which files are you editing?
>
> Edwin.
>

**Comment by NobbZ** [comment]
Something came to my mind right now, might this happen because I'm using a cabal sandbox? `cabal build` by the way does only recompile the changed bits.

**Comment by edwinb** [comment]
Could be, I don't use a sandbox myself. Anyone else noticed this?

In any case, this is highly unlikely to be a problem with Idris' build system - if anything, it is more likely to be a problem with cabal.

**Comment by NobbZ** [comment]
If we can nail it down to either sandboxing or a specific cabal version, I
will do a bugreport over there.
Am 02.01.2015 20:46 schrieb "Edwin Brady" <notifications@github.com>:

**Comment by NobbZ** [comment]
OK, nailed it down to be not even `cabal`s fault (in the first place) :)

The following happened:
* I opted for building haddock for every installed package in cabal-config
* I also opted for using all my cores in cabal-config
* Cabal did not show any progress (despite it is working on idris) du to multiple jobs
* Since there was no progress report I did not see, that cabal was invoking haddock everytime I built idris

After changing this settings back to default I was able to build idris in a couple of seconds, as expected. Anyway, thank you for pointing me to searching the issue with cabal instead of your build script.

So we can close this one.
